### DISCUSSION TITLE
Adding configuration to Bundler plugin

### DISCUSSION THREAD

**Discussion by okuramasafumi** [ruby/rubygems#5736] [Q&A] [answered]
Hi all,

I'm creating a Bundler plugin reading [the documentation](https://bundler.io/guides/bundler_plugins.html).
https://github.com/okuramasafumi/bundler-ctags_generator
I already built a basic functionality but I'd also like to add configuration to the plugin.
How can I achieve that? The obvious way is to use environment variables but I'd rather want to avoid it.
Maybe the best is to use files like `gemrc`?

`gemrc` is not a Ruby file, so maybe we need `bundlerc`?

**Comment by hsbt** [comment, marked as answer]
`Bundler.settings` already reads namespaced keys, so a plugin does not need a config file of its own. `bundle config set --local ctags_generator.paths lib,app` writes the value into `.bundle/config`, and `Bundler.settings["ctags_generator.paths"]` reads it back. That gives you per-project configuration with no environment variable, and the same key still works as `BUNDLE_CTAGS_GENERATOR__PATHS` for anyone who does want to pass it through the environment.
